packages/samples/headless-ssr/src/components/result-list.tsx

```diff
@@ -1,18 +1,39 @@
-'use client';
+import {
+  ResultList as ResultListController,
+  ResultListState,
+} from '@coveo/headless';
+import {useEffect, useState, FunctionComponent} from 'react';
 
-import {Result} from '@coveo/headless';
+interface ResultListProps {
+  initialState: ResultListState;
+  controller?: ResultListController;
+}
+
+export const ResultList: FunctionComponent<ResultListProps> = (props) => {
```

Comment:
I copied most of this from the existing `result-list.fn.tsx` sample in `headless-react-samples`, except:
* I made the `controller` prop optional.
* I added the `initialState` prop.
* I removed `ResultLink` and `QuickView`.

Reply:
Changes here are necessary in order to subscribe to changes.